packages/astro/src/runtime/server/render/astro/render.ts

```diff
@@ -165,3 +166,101 @@ async function bufferHeadContent(result: SSRResult) {
 		}
 	}
 }
+
+export async function renderToAsyncIterable(
```

Comment:
I am a bit worried that we don't have any comments that explain what this function is doing. We should explain both high-level and low-level stuff, so other developers can understand the logic in case there's some bugfix to do.

Reply:
Added comments throughout the function and tried to explain how it works. Also happy to explain on a call if you'd like!